### ISSUE TITLE
Discovered entities with a name that starts with the device name

### ISSUE THREAD

**Issue by PimDoos** [adorobis/hacomfoairmqtt#43]
We currently do not follow Home Assistant's entity naming best practices. This causes an error in HA 2023.08 and will break in HA 2024.2.

> Some MQTT entities have an entity name that starts with the device name. This is not expected. To avoid a duplicate name the device name prefix is stripped of the entity name as a work-a-round. Please inform the maintainer of the software application that supplies the affected entities to fix this issue.
> List of affected entities:
> sensor.comfoair_exhaust_temperature
> sensor.comfoair_ewt_temperature
> sensor.comfoair_supply_fan_speed
> button.comfoair_reset_filter
> (continued)

**Comment by PimDoos** [comment]
[The naming of MQTT entities changes to correspond with HA guidelines](https://developers.home-assistant.io/blog/2023-057-21-change-naming-mqtt-entities/)
TLDR we need to set `has_entity_name` to True in device discovery and remove the device name prefix from the entity name.

I'll throw in a PR.
